packages/dialect-react/components/ApiContext/index.tsx

```diff
@@ -109,6 +110,7 @@ export const ApiProvider = ({ dapp, children }: PropsType): JSX.Element => {
 
   useEffect(() => {
     if (isWalletConnected) {
+      removeToken();
```

Comment:
just out of curiosity, why not remove the token when the wallet gets disconnected rather than when connected?

Reply:
make sense. But also anyway if wallet changed isWalletConnected will be trigger and old token need to be removed. But I think we can move this line above this condition

Reply:
@fsher  what do you think?